src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentAction.java

```diff
@@ -833,7 +840,51 @@ public ActionForward approve(ActionMapping mapping, ActionForm form, HttpServlet
         if (StringUtils.isNotBlank(iWantDocForm.getNewAdHocRoutePerson().getId())) {
             insertAdHocRoutePerson(mapping, iWantDocForm, request, response);
         }
-        return super.approve(mapping, form, request, response);
+        
+        IWantDocument iwantDoc = iWantDocForm.getIWantDocument();
+        boolean shouldAddNote = getFinancialSystemWorkflowHelperService().isAdhocApprovalRequestedForPrincipal(
+                iwantDoc.getDocumentHeader().getWorkflowDocument(), GlobalVariables.getUserSession().getPrincipalId()) && 
+                StringUtils.isNotBlank(iwantDoc.getCompleteOption());
+        LOG.debug("approve, shouldAddNote: {}", shouldAddNote);
+
+        ActionForward forward = super.approve(mapping, form, request, response);
+        
+        if (shouldAddNote) {
+            addCompleteOptionNote(iwantDoc);
+        }
+        
+        return forward;
+    }
+
+    private void addCompleteOptionNote(IWantDocument iwantDoc) {
+        String noteText;
+        Person loggedInUser = GlobalVariables.getUserSession().getPerson();
+        if (StringUtils.equalsIgnoreCase(iwantDoc.getCompleteOption(), "Y")) {
+            noteText = MessageFormat.format(
+                    getConfigurationService().getPropertyValueAsString(CUPurapKeyConstants.MESSAGE_IWANT_DOCUMENT_APPROVE_FINALIZED),
+                    loggedInUser.getPrincipalName());
+        } else {
+            noteText = MessageFormat.format(
+                    getConfigurationService().getPropertyValueAsString(CUPurapKeyConstants.MESSAGE_IWANT_DOCUMENT_APPROVE_SUBMIT_TO_WORKFLOW),
+                    loggedInUser.getPrincipalName());
+        }
+        
+        Note note = getDocumentService().createNoteFromDocument(iwantDoc, noteText);
+        note.setAuthorUniversalIdentifier(loggedInUser.getPrincipalId());
```

Comment:
I believe the note author is automatically being set up by the DocumentService call and/or its downstream NoteService call. You may not need to set it explicitly.

Also, should the system user be configured as the note author instead, like what was done in another recent IWantDoc add-note feature? You can use `GlobalVariables.doInNewGlobalVariables()` to temporarily change the configured user if necessary; The #1451 PR has an example.

Reply:
I will ask Cathy on the user story, she provided the note texts.